qiskit_nature/second_q/hamiltonians/lattices/kagome_lattice.py

```diff
@@ -0,0 +1,305 @@
+                Defaults to 0.0.
+            boundary_condition: Boundary condition for the lattice.
+                The available boundary conditions are:
+                BoundaryCondition.OPEN, BoundaryCondition.PERIODIC.
+                Defaults to BoundaryCondition.OPEN.
+
+        Raises:
+            ValueError: Given size, edge parameter or boundary condition are invalid values.
+        """
+        self.rows = rows
+        self.cols = cols
+        self.size = (rows, cols)
+        self.dim = 2
+        self.boundary_condition = boundary_condition
+        self.num_sites_per_cell = 3
+        self.cell_positions = np.array([[0, 0], [1, 0], [1 / 2, np.sqrt(3) / 2]])
+        self.basis = np.array([[2, 0], [1, np.sqrt(3)]])
+
+        self.edge_parameter = edge_parameter
+        self.onsite_parameter = onsite_parameter
+
+        graph = PyGraph(multigraph=False)
+        graph.add_nodes_from(range(self.num_sites_per_cell * np.prod(self.size)))
+
+        # add edges excluding the boundary edges
+        bulk_edges = self._bulk_edges()
+        graph.add_edges_from(bulk_edges)
+
+        # add self-loops
+        self_loop_list = self._self_loops()
+        graph.add_edges_from(self_loop_list)
+
+        # add edges that cross the boundaries
+        boundary_edge_list = self._boundary_edges()
+        graph.add_edges_from(boundary_edge_list)
+
+        # a list of edges that depend on the boundary condition
+        self.boundary_edges = [(edge[0], edge[1]) for edge in boundary_edge_list]
+        super().__init__(graph)
+        # default position
+        self.pos = self._default_position()
+
+    def draw_without_boundary(
+        self,
+        *,
+        self_loop: bool = False,
+        style: Optional[LatticeDrawStyle] = None,
+    ):
+        r"""Draw the lattice with no edges between the boundaries.
+
+        Args:
+            self_loop: Draw self-loops in the lattice. Defaults to False.
+            style : Styles for rustworkx.visualization.mpl_draw.
+                Please see
+                https://qiskit.org/documentation/rustworkx/stubs/rustworkx.visualization.mpl_draw.html#rustworkx.visualization.mpl_draw
+                for details.
+        """
+        graph = self.graph
+        num_sites_per_cell = self.num_sites_per_cell
+        size = self.size
+
+        if style is None:
+            style = LatticeDrawStyle()
+        elif not isinstance(style, LatticeDrawStyle):
+            style = LatticeDrawStyle(**style)
+
+        if style.pos is None:
+            style.pos = {}
+            for cell_idx in range(np.prod(size)):
+                cell_coord = np.array(divmod(cell_idx, size[0])[::-1])
+
+                for i in range(3):
+                    node_i = num_sites_per_cell * cell_idx + i
+                    style.pos[node_i] = (
+                        np.dot(cell_coord, self.basis) + self.cell_positions[i]
+                    ).tolist()
```

Comment:
Except for this section, this method is identical to `HyperCubicLattice.draw_without_boundary`. Can we avoid this code duplication by creating a private method which is called when `if style.pos is None`. It could look something like this:
```python
def _style_pos(self, ...) -> dict:
    # do logic
    # return dict
    
# and then it is called like:
if style.pos is None:
    style.pos = self._style_pos()
```
The same refactoring can be done to the `TriangularLattice` class

Reply:
Since there would be a lot of resemblance between `_default_position` and `_style_pos`, we can add a keyboard argument `with_boundaries` to `_default_position` like so:

```python
def _default_position(self, with_boundaries: bool = True) -> Dict[int, List[float]]:
    ...
    if with_boundaries and boundary_condition == BoundaryCondition.PERIODIC:
        width = 0.2
        ...
```

and reuse it in `draw_without_boundary` with `with_boundaries` set to `False`?